### PR TITLE
adding model and controller for group

### PR DESCRIPTION
**Pull request by velimd**

### CODE REVIEW

File: documentation/swagger/api.yaml [hunk 1]
```diff
@@ -290,6 +290,33 @@ paths:
...
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Name of Organisation"
+          required: true
+          schema:
+            $ref: "#/definitions/Organisation"
```
Comment (linusnorton) [inline documentation/swagger/api.yaml]: probably Group

File: documentation/swagger/api.yaml [hunk 2]
```diff
@@ -290,6 +290,33 @@ paths:
...
+      summary: "Create an group"
+      description: ""
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Name of Organisation"
```
Comment (linusnorton) [inline documentation/swagger/api.yaml]: Also Group

File: src/group/GroupView.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as chai from "chai";
+import { GroupView } from "./GroupView";
+
+describe("GroupView", () => {
+    const view = new GroupView(
+        {
+            1: { id: 1, name: "org", scheme_id: 1},
```
Comment (linusnorton) [inline src/group/GroupView.spec.ts]: probably group as the name rather than org

File: src/group/GroupView.spec.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            name: "group",
+            organisation_id: 2
+        });
+
+        chai.expect(actual.id).to.equal("/group/" + 1);
+        chai.expect(actual.name).to.equal("group");
+        chai.expect(actual.organisation).to.equal("/organisation/" + 2);
+    });
+
+    it("populates the links object", async () => {
```
Comment (linusnorton) [inline src/group/GroupView.spec.ts]: I think we need an extra test here - what shape are the organisation objects in? They should be OrganisationJsonView which means your GroupView will need to take a OrganisationView to create it, fun fun

File: src/group/GroupView.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        private readonly organisations: OrganisationIndex
+    ) { }
+
+    /**
+     * Return the organisation ViewModel and add the scheme to the links
+     */
+    public create(links: object, group: NonNullId<Group>): GroupJsonView {
+        const organisationId = fromOrganisationId(group.organisation_id);
+
+        links[organisationId] = links[organisationId] || this.organisations[group.organisation_id];
```
Comment (linusnorton) [inline src/group/GroupView.ts]: As noted above you will probably need to run `this.organisations[group.organisation_id];` through the OrganisationView to get the proper JSON and links added.

File: src/group/GroupViewFactory.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { ViewFactory } from "../service/controller/GenericGetController";
+import { Group, GroupJsonView } from "./Group";
+import { GenericRepository } from "../database/GenericRepository";
+import { Organisation } from "../organisation/Organisation";
+import { GroupView } from "./GroupView";
+
+/**
+ * Creates an OrganisationViewFactory
```
Comment (linusnorton) [inline src/group/GroupViewFactory.ts]: GroupView

File: src/group/command/CreateGroupCommand.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { GenericRepository } from "../../database/GenericRepository";
+import { Group } from "../Group";
+import { OrganisationId } from "../../organisation/Organisation";
+
+/**
+ * Command that will create an organisation
```
Comment (linusnorton) [inline src/group/command/CreateGroupCommand.ts]: Group

File: documentation/swagger/api.yaml (hunk 1 above)
Comment (velimd) [reply documentation/swagger/api.yaml]: why is it for parameters in organisation post it says scheme,
I thought this was create as one of the parameter is organisation for group.

File: documentation/swagger/api.yaml (hunk 2 above)
Comment (velimd) [reply documentation/swagger/api.yaml]: same as above

File: src/group/GroupView.spec.ts (hunk 3 above)
Comment (velimd) [reply src/group/GroupView.spec.ts]: but this is taking a organisation object which is then used to link it in the group

**Pull request merged**